src/main/java/org/threadly/litesockets/ThreadedSocketExecuter.java

```diff
@@ -94,31 +87,46 @@ protected void startupService() {
 
   @Override
   protected void shutdownService() {
-    localAcceptScheduler.shutdownNow();
-    localReadScheduler.shutdownNow();
-    localWriteScheduler.shutdownNow();
-    closeSelector(acceptSelector);
-    closeSelector(readSelector);
-    closeSelector(writeSelector);
-    clients.clear();
-    servers.clear();
-  }
-
-  @Override
-  public void flagNewWrite(Client client) {
-    ArgumentVerifier.assertNotNull(client, "Client");
-    if(isRunning() && clients.containsKey(client.getChannel()) && client.canWrite()) {
-      writeScheduler.execute(new AddToSelector(client, writeSelector, SelectionKey.OP_WRITE));
-      writeSelector.wakeup();
+    for(final Client client: clients.values()) {
+      client.close();
     }
-  }
-
+    for(final Server server: servers.values()) {
+      server.close();
+    }
+    closeSelector(localAcceptScheduler, acceptSelector);
+    closeSelector(localReadScheduler, readSelector);
+    closeSelector(localWriteScheduler, writeSelector);
+  }  
+  
   @Override
-  public void flagNewRead(Client client) {
+  public void setClientOperations(final Client client) {
     ArgumentVerifier.assertNotNull(client, "Client");
-    if(isRunning() && clients.containsKey(client.getChannel()) && client.canRead()) {
-      readScheduler.execute(new AddToSelector(client, readSelector, SelectionKey.OP_READ));
+    if(isRunning() && !clients.containsKey(client.getChannel()) && !client.isClosed() && client.getClientsSocketExecuter() == this) {
```

Comment:
hope it's not annoying, but so it does not get lost there is another putIfAbsent weirdness